src/renderer/widgets/RemoveProxy/model/remove-proxy-model.ts

```diff
@@ -0,0 +1,209 @@
+import { UnsignedTransaction } from '@substrate/txwrapper-polkadot';
+import { combine, createEvent, createStore, sample } from 'effector';
+import { spread } from 'patronum';
+import { ApiPromise } from '@polkadot/api';
+
+import { Account, Chain, HexString, ProxyAccount, ProxyGroup } from '@shared/core';
+import { accountUtils, walletModel, walletUtils } from '@entities/wallet';
+import { dictionary } from '@shared/lib/utils';
+import { getSignatoryAccounts } from '@pages/Operations/common/utils';
+import { proxyModel } from '@entities/proxy';
+
+export const enum Step {
+  CONFIRMATION,
+  SIGNING,
+  SUBMIT,
+}
+
+type FlowStartedProps = {
+  chain: Chain;
+  proxyAccount: ProxyAccount;
+};
+const flowStarted = createEvent<FlowStartedProps>();
+
+const $chain = createStore<Chain | null>(null);
+const $proxyAccount = createStore<ProxyAccount | null>(null);
+const $activeStep = createStore<Step>(Step.CONFIRMATION).reset(flowStarted);
+const $signatory = createStore<Account | null>(null).reset(flowStarted);
+
+const $unsignedTx = createStore<UnsignedTransaction | null>(null).reset(flowStarted);
+const $signature = createStore<HexString | null>(null).reset(flowStarted);
+
+const $proxiedAccount = combine(
+  {
+    wallets: walletModel.$wallets,
+    accounts: walletModel.$accounts,
+    proxyAccount: $proxyAccount,
+  },
+  ({ wallets, accounts, proxyAccount }) => {
+    if (!proxyAccount) return null;
+
+    const walletsMap = dictionary(wallets, 'id');
+
+    return (
+      accounts.find(
+        (a) => a.accountId === proxyAccount.proxiedAccountId && !walletUtils.isWatchOnly(walletsMap[a.walletId]),
+      ) || null
+    );
+  },
+  {
+    skipVoid: false,
+  },
+);
+
+const $proxiedWallet = combine(
+  {
+    wallets: walletModel.$wallets,
+    proxiedAccount: $proxiedAccount,
+  },
+  ({ wallets, proxiedAccount }) => {
+    if (!proxiedAccount) return null;
+
+    return wallets.find((w) => w.id === proxiedAccount.walletId) || null;
+  },
+  {
+    skipVoid: false,
+  },
+);
+
+const $signatories = combine(
+  {
+    wallets: walletModel.$wallets,
+    accounts: walletModel.$accounts,
+    proxiedAccount: $proxiedAccount,
+    chain: $chain,
+  },
+  ({ wallets, accounts, proxiedAccount, chain }) => {
+    if (!chain || !proxiedAccount) return null;
+
+    return accountUtils.isMultisigAccount(proxiedAccount)
+      ? getSignatoryAccounts(accounts, wallets, [], proxiedAccount.signatories, chain.chainId)
+      : null;
+  },
+  {
+    skipVoid: false,
+  },
+);
+
+const signatorySelected = createEvent<Account>();
```

Comment:
Better to group events before stores

Reply:
Agree, link to our [tg](https://t.me/c/1926452604/963) discussion regarding grouping of events and stores